### PR TITLE
new kalman filter on quadcopter for altitude hold

### PR DESCRIPTION
**Pull request by pdavid747**
## Main Changes
New Kalman Filter to include data for Barometer and GNSS altitude. this will be used for altitude control.

## Tests

This branch still needs to be tested.

### CODE REVIEW

File: design/Parts_for_3D_printing_and_laser_cutting/Frame_parts/Quadcopter_base_suport Drawing v5.cdr
Comment (alphaville) [inline design/Parts_for_3D_printing_and_laser_cutting/Frame_parts/Quadcopter_base_suport Drawing v5.cdr:1]: @pdavid747 @jamie-54 I see that very often the files in `design` get updated in some branches. Why does this happen? My guess is that you branch out of stale versions of `dev`. Next time, try:

```
git checkout dev
git pull origin dev
git checkout -b feature/something dev
```

Also, please give descriptive titles to the PRs ("Fix/new kalman filter" is not a descriptive title? Is this simulations? A whitepaper? Also what is the meaning of "new"?"

**Comment by jamie-54** [comment]
@alphaville @Runway27 this PR is ready for review

**Comment by jamie-54** [comment]
@alphaville @ejb-11 @Runway27  are we happy for this to be merged?

**Review by Runway27 — APPROVED** [review]
Looks good to me

**Pull request merged**